velox/functions/prestosql/tests/ParsePrestoDataSizeTest.cpp

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <velox/common/base/VeloxException.h>
+#include <velox/vector/SimpleVector.h>
+#include <optional>
+#include <utility>
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using facebook::velox::functions::test::FunctionBaseTest;
+
+namespace facebook::velox {
+namespace {
+
+class ParsePrestoDataSizeTest : public FunctionBaseTest {
+ protected:
+  void assertVectors(
+      const std::string& expression,
+      const VectorPtr& arg0,
+      const VectorPtr& expected) {
+    auto result = evaluate(expression, makeRowVector({arg0}));
+    test::assertEqualVectors(expected, result);
+  }
+
+  void assertExpression(std::string input, const std::string& outputStr) {
```

Comment:
nit: you arent asserting a general expression here ; you are asserting the size , maybe renamed this function to assertSize ?